examples/supabase/src/index.ts

```
@@ -1,28 +1,43 @@
-
 import '../public/style.css';
 
 import { startReplication } from './replication';
 import { createDatabase } from './database';
 import { addClickHandlers } from './handlers';
 
-const heroesList = document.querySelector('#heroes-list');
+window.addEventListener('unhandledrejection', (e) => {
+    console.error(
+        `got unhandled error`,
+        e.reason.message,
+        JSON.stringify({
+            // @ts-expect-error ignore
+            frame: e.currentTarget?.ifrm?.id,
+        }),
+        e
+    );
+});
```

It was difficult to get to the bottom of the issue, because the page was throwing unhandled exceptions and test cafe only captures logs, not exceptions. So the test was effectively silently failing, and test cafe was just going into a loop trying to get it to pass until the test finally timed out.

Now if there are any unhandled exceptions on the page, it's logged as an error and the test cafe tests will immediately stop and print out the relevant info - should make future problems much easier to debug.